llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -9393,6 +9393,10 @@ static ConstantRange getRangeForIntrinsic(const IntrinsicInst &II) {
     if (!II.getParent() || !II.getFunction())
       break;
     return getVScaleRange(II.getFunction(), Width);
+  case Intrinsic::scmp:
+  case Intrinsic::ucmp:
+    return ConstantRange::getNonEmpty(APInt::getAllOnes(Width),
+                                      APInt::getOneBitSet(Width, 0));
```

Comment:
```suggestion
                                      APInt(Width, 2));
```
The upper bound is exclusive, so the range should be `[-1, 2)`, while you're currently constructing `[-1, 1)`.

Reply:
Thanks, fixed.